pkg/spec/spec.go

```diff
@@ -533,11 +534,27 @@ func addRlimits(config *CreateConfig, g *generate.Generator) error {
 	// If not explicitly overridden by the user, default number of open
 	// files and number of processes to the maximum they can be set to
 	// (without overriding a sysctl)
-	if !nofileSet && !isRootless {
-		g.AddProcessRlimits("RLIMIT_NOFILE", kernelMax, kernelMax)
+	if !nofileSet {
+		if isRootless {
+			var rlimit unix.Rlimit
+			if err := unix.Getrlimit(unix.RLIMIT_NOFILE, &rlimit); err != nil {
```

Comment:
I don't think we can hard-error on these; I can see cases where this won't work, but the rest of Podman will function fine.

Reply:
Switched to warning.

Reply:
we have `--ulimit host` that tells Podman to not set any limit block in the OCI configuration file.  When there are no limit sets the OCI runtime won't apply any limit to the container so the host settings are propagated to the container.

Do we have to make that the default (AFAICS it is not yet supported by Buildah)?

Reply:
I would figure it was the default.  Where does the 1024,1024 come from that we have now?
I also found it curious that 

$ podman unshare grep open /proc/self/limits 
Max open files            524288               524288               files     

Why is the default to set the soft limit== harlimit in a user namespace?

Reply:
I think we do that automatically to avoid the `max number of open files` error when binding too many ports: https://github.com/containers/libpod/commit/9126b45a3f2255b103ef60f305f8ca9ab2805b50.


The issue with rootless is that we cannot set higher limits that the user currently has, so we can't default to the values used by root as they could not be allowed in the current session.

At the same time, we risk of setting higher limits than a root container would have if defaulting to `--ulimit host`.

We'd probably have to check each default value and lower it when it is higher than what the user is allowed to use, but we can adress this part later

Reply:
So should we merge this change or drop it?